tests/integration/routes/test_feedback.py

```diff
@@ -0,0 +1,137 @@
+from app import settings
+from tests.integration.integration_test_case import IntegrationTestCase
+
+
+class TestFeedback(IntegrationTestCase):
+    def test_questionnaire_not_completed(self):
+        # Given I launch the test_feedback questionnaire
+        self.launchSurvey("test_feedback")
+
+        # When I try to view the feedback page without completing the questionnaire
+        self._get_feedback_send()
+
+        # Then I get shown a 404 error
+        self.assertStatusNotFound()
+
+    def test_questionnaire_not_completed_post(self):
+        # Given I launch the test_feedback questionnaire
+        self.launchSurvey("test_feedback")
+
+        # When I try to POST to the feedback page without completing the questionnaire
+        self.post(url="/submitted/feedback/send")
```

Comment:
These all should use a top level var instead of repeated the url.

Reply:
Or they could use url for